tests/src/core/testqgsmaprotation.cpp

```diff
@@ -0,0 +1,219 @@
+  // the actual size doesn't matter as QgsRenderChecker will
+  // re-set it to the size of the expected image
+  mMapSettings.setOutputSize( QSize(256,256) );
+
+  mReport += "<h1>Map Rotation Tests</h1>\n";
+}
+
+//runs after all tests
+void TestQgsMapRotation::cleanupTestCase()
+{
+  QgsApplication::exitQgis();
+
+  // TODO: delete layers (or is it done by exitQgis ?)
+
+  QString myReportFile = QDir::tempPath() + QDir::separator() + "qgistest.html";
+  QFile myFile( myReportFile );
+  if ( myFile.open( QIODevice::WriteOnly | QIODevice::Append ) )
+  {
+    QTextStream myQTextStream( &myFile );
+    myQTextStream << mReport;
+    myFile.close();
+  }
+}
+
+void TestQgsMapRotation::rasterLayer()
+{
+  mMapSettings.setLayers( QStringList() << mRasterLayer->id() );
+  mMapSettings.setExtent( mRasterLayer->extent() );
+  mMapSettings.setRotation( 45 );
+  // This ensures rotated image is all visible by tweaking scale
+  mMapSettings.setExtent( mMapSettings.visibleExtent() );
+  QVERIFY( render( "raster+45" ) );
+
+  mMapSettings.setRotation( -45 );
+  QVERIFY( render( "raster-45" ) );
+}
+
+void TestQgsMapRotation::pointsLayer()
+{
+  mMapSettings.setLayers( QStringList() << mPointsLayer->id() );
+
+  // SVG points, fixed (no) rotation
+  QString qml = mTestDataDir + "points.qml";
+  bool success = false;
+  mPointsLayer->loadNamedStyle( qml, success );
+  QVERIFY( success );
+  mMapSettings.setExtent( QgsRectangle(-105.5,37,-97.5,45) );
+  mMapSettings.setRotation( -60 );
+  QVERIFY( render( "svgpoints-60" ) );
+
+  // SVG points, data defined rotation
+  qml = mTestDataDir + "points_single_symbol_datadefined_rotation.qml";
+  success = false;
+  mPointsLayer->loadNamedStyle( qml, success );
+  QVERIFY( success );
+  mMapSettings.setExtent( QgsRectangle(-116,33,-107,42) );
+  mMapSettings.setRotation( 90 );
+  QVERIFY( render( "svgpoints-datadefined+90" ) );
+
+  // TODO: SVG points, fixed (defined) rotation ?
+
+  // Simple points, data defined rotation
+  qml = mTestDataDir + "points_single_symbol.qml";
+  success = false;
+  mPointsLayer->loadNamedStyle( qml, success );
+  QVERIFY( success );
+  mMapSettings.setExtent( QgsRectangle(-116,33,-107,42) );
+  mMapSettings.setRotation( 90 );
+  QVERIFY( render( "simplepoints-datadefined+90" ) );
+
+  // Simple points, fixed (no) rotation
+  qml = mTestDataDir + "points_graduated_symbol.qml";
+  success = false;
+  mPointsLayer->loadNamedStyle( qml, success );
+  QVERIFY( success );
+  mMapSettings.setExtent( QgsRectangle(-108,26,-100,34) );
+  mMapSettings.setRotation( 30 );
+  QVERIFY( render( "simplepoints+30" ) );
+
+  // TODO: simple points, fixed (defined) rotation ?
+}
+
+void TestQgsMapRotation::linesLayer()
+{
+  mMapSettings.setLayers( QStringList() << mLinesLayer->id() );
+
+  // Arrowed line with parallel labels
+  QString qml = mTestDataDir + "lines_cardinals_arrowed_parallel_label.qml";
+  bool success = false;
+  mLinesLayer->loadNamedStyle( qml, success );
```

Comment:
@strk it's this test which is failing - I think you've missed the associated qml file from the PR?